week4/filter-less/helpers.c

```diff
@@ -0,0 +1,132 @@
+#include "helpers.h"
+#include "math.h"
+
+void grayscale(int height, int width, RGBTRIPLE image[height][width])
+{
+    for (int i = 0; i < height; i++)
+    {
+        for (int j = 0; j < width; j++)
+        {
+            float Red = image[i][j].rgbtRed;
+            float Green = image[i][j].rgbtGreen;
+            float Blue = image[i][j].rgbtBlue;
+
+            int average = round((Red + Green + Blue) / 3);
+            image[i][j].rgbtRed = image[i][j].rgbtGreen = image[i][j].rgbtBlue = average;
+        }
+    }
+
+    return;
+}
+
+void sepia(int height, int width, RGBTRIPLE image[height][width])
+{
+    for (int i = 0; i < height; i++)
+    {
+        for (int j = 0; j < width; j++)
+        {
+            float originalRed = image[i][j].rgbtRed;
+            float originalGreen = image[i][j].rgbtGreen;
+            float originalBlue = image[i][j].rgbtBlue;
+
+            float sepiaRed = round(.393 * originalRed + .769 * originalGreen + .189 * originalBlue);
+            float sepiaGreen = round(.349 * originalRed + .686 * originalGreen + .168 * originalBlue);
+            float sepiaBlue = round(.272 * originalRed + .534 * originalGreen + .131 * originalBlue);
+
+            if (sepiaRed > 255)
+            {
+                sepiaRed = 255;
+            }
+
+            if (sepiaGreen > 255)
+            {
+                sepiaGreen = 255;
+            }
+
+            if (sepiaBlue > 255)
+            {
+                sepiaBlue = 255;
+            }
+
+            image[i][j].rgbtRed = sepiaRed;
+            image[i][j].rgbtGreen = sepiaGreen;
+            image[i][j].rgbtBlue = sepiaBlue;
+        }
+    }
+
+    return;
+}
+
+void reflect(int height, int width, RGBTRIPLE image[height][width])
+{
+    for (int i = 0; i < height; i++)
+    {
+        for (int j = 0; j < width / 2; j++)
+        {
+            RGBTRIPLE temp = image[i][j];
+            image[i][j] = image[i][width - (j + 1)];
+            image[i][width - (j + 1)] = temp;
+        }
+    }
+
+    return;
+}
+
+void blur(int height, int width, RGBTRIPLE image[height][width])
+{
+    RGBTRIPLE copy[height][width];
+
+    for (int i = 0; i < height; i++)
+    {
+        for (int j = 0; j < width; j++)
+        {
+            copy[i][j] = image[i][j];
+        }
+
+        for (int i = 0; i < height; i++)
```

Comment:
- You use the same variable inside a nested scope. This is called variable shadowing and should be avoided.
- Your `blur` function is rather long and has many for-loops. You could move some of your code to helper functions with meaningful names, to increase the understandability of your code.